src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlQueryMetadataCache.cs

```diff
@@ -228,21 +234,27 @@ internal void AddQueryMetadata(SqlCommand sqlCommand, bool ignoreQueriesWithRetu
             }
 
             // By default evict after 10 hours.
-            _cache.Set(cacheLookupKey, ciperMetadataDictionary, DateTimeOffset.UtcNow.AddHours(10));
+            _cache.Set(cacheLookupKey, cipherMetadataDictionary, DateTimeOffset.UtcNow.AddHours(10));
+            if (sqlCommand.requiresEnclaveComputations)
+            {
+                Dictionary<int, SqlTceCipherInfoEntry> keysToBeCached = CreateCopyOfEnclaveKeys(sqlCommand.keysToBeSentToEnclave);
+                _cache.Set(enclaveLookupKey, keysToBeCached, DateTimeOffset.UtcNow.AddHours(10));
+            }
         }
 
         /// <summary>
         /// <para> Remove the metadata for a specific query from the cache.</para>
         /// </summary>
         internal void InvalidateCacheEntry(SqlCommand sqlCommand)
         {
-            string cacheLookupKey = GetCacheLookupKeyFromSqlCommand(sqlCommand);
+            (string cacheLookupKey, string enclaveLookupKey) = GetCacheLookupKeysFromSqlCommand(sqlCommand);
```

Comment:
Are value tuples supported on all the targets for this library? I'd love to be able to use them but I thought they needed an additional package on netfx and possible older netcore and netstandard.

Reply:
All of them except for netfx, unfortunately.